File: twitterbot2.py
import tweepy
import time

# Insert your own API keys here
consumer_key = "your_consumer_key"
consumer_secret = "your_consumer_secret"
access_token = "your_access_token"
access_token_secret = "your_access_token_secret"

# Authenticate with Twitter API
auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)

# Initialize API object
api = tweepy.API(auth)

# Define a function to like tweets based on search query
def like_tweets(search_query):
    for tweet in tweepy.Cursor(api.search, search_query).items(10):
        try:
            tweet.favorite()
            print("Liked tweet by:", tweet.user.screen_name)
        except tweepy.TweepError as e:
            print(e.reason)
        except StopIteration:
            break

# Define a function to comment on tweets based on search query
def comment_on_tweets(search_query, comment_text):
    for tweet in tweepy.Cursor(api.search, search_query).items(10):
        try:
            tweet.retweet()
            tweet.favorite()
            print("Commented on tweet by:", tweet.user.screen_name)
        except tweepy.TweepError as e:
            print(e.reason)
        except StopIteration:
            break

# Define a function to follow accounts based on search query
def follow_accounts(search_query):
    for tweet in tweepy.Cursor(api.search, search_query).items(10):
        try:
            if tweet.user.followers_count > 1000:
                tweet.user.follow()
                print("Followed", tweet.user.screen_name)
        except tweepy.TweepError as e:
            print(e.reason)
        except StopIteration:
            break

# Define a function to unfollow accounts with less than 1000 followers
def unfollow_accounts_less_than_1000():
    for friend in tweepy.Cursor(api.friends).items():
        try:
            if friend.followers_count < 1000:
                friend.unfollow()
                print("Unfollowed", friend.screen_name)
        except tweepy.TweepError as e:
            print(e.reason)
        except StopIteration:
            break

# Define a function to unfollow accounts that don't follow back in less than 3 days
def unfollow_non_followers():
    for friend in tweepy.Cursor(api.friends).items():
        try:
            if (time.time() - friend.status.created_at.timestamp()) > (3 * 24 * 3600):
                friend.unfollow()
                print("Unfollowed", friend.screen_name)
        except tweepy.TweepError as e:
            print(e.reason)
        except StopIteration:
            break

# Call

File: twitterbot2.js
const Twit = require('twit');

const T = new Twit({
  consumer_key: 'YOUR_CONSUMER_KEY',
  consumer_secret: 'YOUR_CONSUMER_SECRET',
  access_token: 'YOUR_ACCESS_TOKEN',
  access_token_secret: 'YOUR_ACCESS_TOKEN_SECRET',
  timeout_ms: 60 * 1000,  // optional HTTP request timeout to apply to all requests.
});

// Like a tweet
T.post('favorites/create', { id: 'TWEET_ID' }, (err, data, response) => {
  console.log(data);
});

// Follow a user
T.post('friendships/create', { screen_name: 'SCREEN_NAME' }, (err, data, response) => {
  console.log(data);
});

// Unfollow a user
T.post('friendships/destroy', { screen_name: 'SCREEN_NAME' }, (err, data, response) => {
  console.log(data);
});

// Comment on a tweet
T.post('statuses/update', { status: 'YOUR_COMMENT', in_reply_to_status_id: 'TWEET_ID' }, (err, data, response) => {
  console.log(data);
});

// Unfollow non-followers
T.get('followers/ids', { screen_name: 'SCREEN_NAME' }, (err, data, response) => {
  const followers = data.ids;
  T.get('friends/ids', { screen_name: 'SCREEN_NAME' }, (err, data, response) => {
    const friends = data.ids;
    const nonFollowers = friends.filter(friend => !followers.includes(friend));
    nonFollowers.forEach(nonFollower => {
      T.post('friendships/destroy', { user_id: nonFollower }, (err, data, response) => {
        console.log(data);
      });
    });
  });
});

// Follow accounts with more than 1000 followers
T.get('users/search', { q: 'FOLLOW_KEYWORD', count: 100 }, (err, data, response) => {
  data.forEach(user => {
    if (user.followers_count >= 1000) {
      T.post('friendships/create', { user_id: user.id }, (err, data, response) => {
        console.log(data);
      });
    }
  });
});

// Unfollow accounts with less than 1000 followers
T.get('friends/list', { count: 200 }, (err, data, response) => {
  data.users.forEach(user => {
    if (user.followers_count < 1000) {
      T.post('friendships/destroy', { user_id: user.id }, (err, data, response) => {
        console.log(data);
      });
    }
  });
});
